web/src/composebox_typeahead.ts

```diff
@@ -678,7 +848,15 @@ export function get_candidates(query, input_element) {
         compose_ui.set_code_formatting_button_triggered(false);
         const matcher = get_language_matcher(token);
         const matches = language_list.filter((item) => matcher(item));
-        return sort_results(completing, matches, token);
+        const matches_list: LanguageSuggestion[] = matches.map((language) => ({
+            language,
+            type: "syntax",
+        }));
+        const matches_object: LanguageList = {
+            type: "syntax",
+            list: matches_list,
+        };
+        return sort_results(matches_object, token);
```

Comment:
Oh I should be using the specific sort function here too instead of going to the switch statement in `sort_results` -- I'll do that next time I'm working.